link_linux.go

```diff
@@ -612,6 +612,41 @@ func (h *Handle) LinkSetVfVlanQos(link Link, vf, vlan, qos int) error {
 	return err
 }
 
+// LinkSetVfVlanQosProto sets the vlan, qos and protocol of a vf for the link.
+// Equivalent to: `ip link set $link vf $vf vlan $vlan qos $qos proto $proto`
+func LinkSetVfVlanQosProto(link Link, vf, vlan, qos, proto int) error {
```

Comment:
Given this takes already 5 parameters and who knows if more are to come in future, would it make sense to pass a structure as input parameter?

Reply:
Well, I believe that it is better to have the parameters separated to be consistent with the previous 2 functions (LinkSetVfVlanQos and LinkSetVfVlan). We could have used a structure as input parameter if we would have created a more general function for all vlan options. Currently, the name of the function has a one-to-one relationship with the parameters.